native_client/deepspeech.cc

```diff
@@ -485,22 +485,23 @@ Metadata* ModelState::decode_metadata(vector<float>& logits)
 {
   vector<Output> out = decode_raw(logits);
 
-  Metadata* metadata = (Metadata*)malloc(sizeof (Metadata));
+  Metadata* metadata = new Metadata;
```

Comment:
Could we `std::unique_ptr<>` this ?

Reply:
Should the function signature be returning an std::unique_ptr now? Because that means the C API would then be returning a C++ type.

I tried returning metadata.get() instead but now it crashes when it's freed.

Reply:
If you make it a unique_ptr you have to return `metadata.release()`: https://en.cppreference.com/w/cpp/memory/unique_ptr/release